pkg/prometheus/agent/operator.go

```diff
@@ -1184,3 +1187,25 @@ func keyToDaemonSetKey(p monitoringv1.PrometheusInterface, key string) string {
 	keyParts := strings.Split(key, "/")
 	return fmt.Sprintf("%s/%s", keyParts[0], fmt.Sprintf("%s-%s", prompkg.Prefix(p), keyParts[1]))
 }
+
+// a fallback CEL validation function for older kubernetes versions when CEL is not available.
+func validateDaemonSetModeSpec(p *monitoringv1alpha1.PrometheusAgent) error {
+
+	if p.Spec.Replicas != nil {
+		return fmt.Errorf("replicas cannot be set when mode is DaemonSet")
+	}
+
+	if p.Spec.Storage != nil {
+		return fmt.Errorf("storage cannot be set when mode is DaemonSet")
+	}
+
+	if p.Spec.Shards != nil && *p.Spec.Shards > 1 {
```

Comment:
I think we decided on `shards` being not-configurable for the Daemonset mode?
```suggestion
	if p.Spec.Shards != nil {
```